### PR TITLE
Completes 200 GET /api

### PR DESCRIPTION
**Pull request by danrfletcher**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -7,8 +7,21 @@ const app = require('../app.js');
...
 
-describe('GET /api/topics', () => {
-    test('200: responds with all topics', () => {
+describe('GET Requests', () => {
+    test('/api 200: responds with a list of all available endpoints', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((res) => {
+            expect(Object.keys(res.body.endpoints)).toHaveLength(3); //This length will need to be changed as more endpoints are added
```
Comment (nc-hev-mag) [inline __tests__/app.test.js:16]: You can get around this problem by requiring in the endpoints.json file, just as you have in your controller, and then expecting that the body of the response matches it

File: app.js [hunk 2]
```diff
@@ -1,10 +1,17 @@
 const express = require('express');
 const app = express();
 
+const { handlePsqErrors, handleCustomErrors, handleServerErrors } = require('./middlewares/error-handlers.js');
 const { getTopics } = require('./controllers/topic-controllers.js');
+const { getEndpoints } = require('./controllers/endpoint-controllers.js');
 
 app.use(express.json());
```
Comment (nc-hev-mag) [inline app.js:8]: It's best practice to only commit code that is necessary/being used. We don't need to use express.json() until we are dealing with a patch/post request. Something to bear in mind going forwards!

File: middlewares/error-handlers.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+exports.handlePsqErrors = (err, req, res, next) => {
+    if (err.code === "22P02") {
+        res.status(400).send({ msg: "bad request"});
+    } else {
+        next(err);
+    }
+};
+
+exports.handleCustomErrors = (err, req, res, next) => {
```
Comment (nc-hev-mag) [inline middlewares/error-handlers.js:9]: same as above here, if we don't have any error tests then we don't have any need to add these yet

**Review by nc-hev-mag — APPROVED** [review]
Hiya Dan - a couple of tips here. 

I would recommend for sure updating your endpoint test as described before merging.It'll save you having to update it all the time 💯 

Going forwards, make sure to question whether the code you're adding is out of habit or actually needed for that ticket 🤔

Once you've done that - good to merge, no further PR necessary

**Comment by danrfletcher** [comment]
Thanks Hev I updated the endpoint as suggested & removed unnecessary code :)

**Pull request merged**